src/dmd/e2ir.d

```diff
@@ -514,7 +514,19 @@ if (!irs.params.is64bit) assert(tysize(TYnptr) == 4);
             e.Eflags |= EFLAGS_variadic;
     }
 
-    if (retmethod == RET.stack)
+    const isCPPCtor = fd && fd.linkage == LINK.cpp && fd.isCtorDeclaration();
+    if (isCPPCtor &&
+        (irs.params.isLinux ||
+         irs.params.isOSX ||
+         irs.params.isFreeBSD ||
+         irs.params.isDragonFlyBSD ||
+         irs.params.isSolaris))
+    {
+        // CPP constructor returns void on Posix
+        // https://itanium-cxx-abi.github.io/cxx-abi/abi.html#return-value-ctor
+        e = el_combine(e, el_copytree(ethis));
```

Comment:
Merely copying the tree causes problems if the tree has side effects.

Reply:
Changed.